api/python/quilt3/api.py

```diff
@@ -162,10 +162,39 @@ def list_packages(registry=None):
     registry_base_path = get_package_registry(fix_url(registry) if registry else None)
 
     named_packages = registry_base_path.rstrip('/') + '/named_packages/'
+    prev_pkg = None
     for path, _ in list_url(named_packages):
         parts = path.split('/')
-        if len(parts) == 3 and parts[2] == 'latest':
-            yield f'{parts[0]}/{parts[1]}'
+        if len(parts) == 3:
+            pkg = f'{parts[0]}/{parts[1]}'
+            # A package can have multiple versions, but we should only return the name once.
+            if pkg != prev_pkg:
+                prev_pkg = pkg
+                yield pkg
+
+
+def list_package_versions(name, registry=None):
+    """Lists versions of a given package.
+
+    Returns a sequence of (version, hash) of a package in a registry.
+    If the registry is None, default to the local registry.
+
+    Args:
+        registry(string): location of registry to load package from.
+
+    Returns:
+        A sequence of tuples containing the named version and hash.
+    """
+    validate_package_name(name)
+
+    registry_base_path = get_package_registry(fix_url(registry) if registry else None)
+
+    package = registry_base_path.rstrip('/') + '/named_packages/' + name + '/'
+    for path, _ in list_url(package):
+        parts = path.split('/')
+        if len(parts) == 1:
+            pkg_hash, _ = get_bytes(package + parts[0])
+            yield parts[0], pkg_hash.decode().strip()
```

Comment:
Current user experience will be:
```
> list(quilt3.list_package_versions("quiltdata/example"))
[
("1552516174", "cf4ffaa6082aef9caa9d5b92a39808db065f894fbb307b8a121637bbd6f46667"),
("1552517308", "7927f4c3cfab5193a9fe383908303c570646af890cabc2a7a34a5ba5251c7be9"),
("latest", "5d4354f610a497d06c2fb53771e0f7277f473c837252c50a1a4b64eed4fdaca2")
]
```

Do you think it will be obvious to users that the version is a unix timestamp? If not maybe we should make that clearer either by returning datetime objects/formatted timestamp strings or by calling it out in the docstring?

Reply:
Well... Technically, we can't assume they're timestamps. Or at least, we'd need to deal with cases when they're not.

Reply:
Only the "latest" would not be a timestamp, right? In any case, it's not critical for this PR.